src/app/qgsclipboard.cpp

```diff
@@ -154,22 +154,38 @@ void QgsClipboard::setSystemClipboard()
   // that just the next call to systemClipboardChanged() should be ignored
   mIgnoreNextSystemClipboardChange = true;
 
-  QString textCopy = generateClipboardText();
-
   QClipboard *cb = QApplication::clipboard();
 
   // Copy text into the clipboard
+  QString textCopy = generateClipboardText();
+  QMimeData *m = new QMimeData();
+  m->setText( textCopy );
+
+  QgsSettings settings;
+  CopyFormat format = AttributesWithWKT;
+  if ( settings.contains( QStringLiteral( "/qgis/copyFeatureFormat" ) ) )
+  {
+    format = static_cast< CopyFormat >( settings.value( QStringLiteral( "qgis/copyFeatureFormat" ), true ).toInt() );
+  }
+
+  if ( format == AttributesWithWKT && mFeatureClipboard.count() < 1000 )
+  {
+    QString htmlCopy = textCopy;
+    htmlCopy.replace( QRegExp( "\n" ), QStringLiteral( "</td></tr><tr><td>" ) );
```

Comment:
This doesn't need to be a regexp replace, does it? A character based replace will be MUCH faster

    htmlCopy.replace( '\n', QStringLiteral(".....") );